### DISCUSSION TITLE
Text color of `ui.chip` should switch to white on dark background

### DISCUSSION THREAD

**Discussion by tjmuenster** [zauberzeug/nicegui#3109] [Ideas / Feature Requests]
### Description

Hello everybody,

thanks for the amazing framework and the huge steps in the recent months!

With the following code, I've stumpled across an problem with the textcolor of the `ui.chip` element. 

``` python
from nicegui import ui

#The color does not matter. It's main purpose is being dark
ui.colors(primary='#002d55')

with ui.header(elevated=False).classes('text-xl'):
        ui.label('label font-bold').classes('font-bold')
        ui.chip('chip text-bold').classes('text-bold')
        ui.chip('icon chip',icon='person')
        ui.icon('person')

ui.chip(text='chip')

ui.chip('chip font-bold').classes(add='font-bold')

ui.chip('chip text-bold').classes(add='text-bold')

ui.chip('chip text-xl').classes(add='text-xl')

ui.chip('icon',icon='person')

ui.label('Text label')

ui.run()
```
The two elements `ui.label` and `ui.icon` inside `ui.header` are correctly displayed with white textcolor, while the two `ui.chip`-elements are not. First I thought, tailwind and quasar are not working together. But as you can see with the other `ui.chip`-elements, its apparently not.

Even more interessting is, that when you disable the `color` property of `.q-chip` in your average webbrowser, the `ui.chip`-elements inside the header do change their textcolor (see attached screenshots), while the other do not.

![color not disabled](https://github.com/zauberzeug/nicegui/assets/7595907/17ac964e-579d-4796-9f5f-0c534dfa152b)

![color disabled](https://github.com/zauberzeug/nicegui/assets/7595907/12362a1f-e080-4502-b83f-3e35c5f48183)

Am I missing something? Thanks in advance for your help and time!

**Comment by falkoschindler** [comment]
Hi @tjmuenster,

If I understand correctly, you'd expect the text color of `ui.chip` to depend on the background color of the _parent_ element like `ui.header`? I'd argue that it should depend on its own background color instead. But at the moment `ui.chip` doesn't adjust its text color based on its background color. You can do that explicitly using the "dark" prop:

```py
ui.colors(primary='#002d55')

with ui.header():
    ui.label('Label')
    ui.chip('Chip').props('dark')

ui.label('Label')
ui.chip('Chip').props('dark')
```
<img width="287" alt="Screenshot 2024-05-16 at 08 13 52" src="https://github.com/zauberzeug/nicegui/assets/5767091/ba82a87f-e94d-4c64-9b2c-dc9c7904ad8b">

**Comment by tjmuenster** [comment]
Hello @falkoschindler ,

I was adding `ui.chip` to the header to see, if its text color would change based on its parent element, like `ui.label` or `ui.icon` do.

I completly agree with you, that the text color of `ui.chip` should depend on its own background color.
